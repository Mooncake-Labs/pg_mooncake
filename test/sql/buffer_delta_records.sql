CREATE TABLE t (a INT) USING columnstore;

INSERT INTO t VALUES (1), (2), (3);
INSERT INTO t VALUES (1), (2), (3);
SELECT COUNT(*) FROM mooncake.delta_update_records;

DELETE FROM t WHERE a = 1;
SELECT COUNT(*) FROM mooncake.delta_update_records;

-- Request to flush buffered records.
SELECT mooncake.dump_delta_update_records();

-- Check buffered records after flush.
SELECT COUNT(*) FROM mooncake.delta_update_records;

-- Check buffered records after flush.
SELECT COUNT(*) FROM mooncake.delta_update_records;

-- Rows inserted inside of an aborted transaction are deleted by postgres.
BEGIN;
INSERT INTO t VALUES (1), (2), (3);
SELECT COUNT(*) FROM mooncake.delta_update_records;
-- Request to flush buffered records inside of an ongoing transaction doesn't flush any records.
SELECT mooncake.dump_delta_update_records();
SELECT COUNT(*) FROM mooncake.delta_update_records;
ROLLBACK;
SELECT COUNT(*) FROM mooncake.delta_update_records;

-- Drop the table after test.
DROP TABLE t;
