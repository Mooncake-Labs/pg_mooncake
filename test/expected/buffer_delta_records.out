CREATE TABLE t (a INT) USING columnstore;
INSERT INTO t VALUES (1), (2), (3);
INSERT INTO t VALUES (1), (2), (3);
SELECT COUNT(*) FROM mooncake.delta_update_records;
 count 
-------
     4
(1 row)

DELETE FROM t WHERE a = 1;
SELECT COUNT(*) FROM mooncake.delta_update_records;
 count 
-------
     5
(1 row)

-- Request to flush buffered records.
SELECT mooncake.dump_delta_update_records();
 dump_delta_update_records 
---------------------------
 
(1 row)

-- Check buffered records after flush.
SELECT COUNT(*) FROM mooncake.delta_update_records;
 count 
-------
     0
(1 row)

-- Check buffered records after flush.
SELECT COUNT(*) FROM mooncake.delta_update_records;
 count 
-------
     0
(1 row)

-- Rows inserted inside of an aborted transaction are deleted by postgres.
BEGIN;
INSERT INTO t VALUES (1), (2), (3);
SELECT COUNT(*) FROM mooncake.delta_update_records;
 count 
-------
     1
(1 row)

-- Request to flush buffered records inside of an ongoing transaction doesn't flush any records.
SELECT mooncake.dump_delta_update_records();
 dump_delta_update_records 
---------------------------
 
(1 row)

SELECT COUNT(*) FROM mooncake.delta_update_records;
 count 
-------
     1
(1 row)

ROLLBACK;
SELECT COUNT(*) FROM mooncake.delta_update_records;
 count 
-------
     0
(1 row)

-- Drop the table after test.
DROP TABLE t;
